include/zephyr/net/net_event.h

```diff
@@ -289,6 +293,23 @@ enum net_event_l4_cmd {
 #define NET_EVENT_L4_DISCONNECTED			\
 	(_NET_EVENT_L4_BASE | NET_EVENT_L4_CMD_DISCONNECTED)
 
+
+/** Event raised when IPv4 network connectivity becomes available. */
+#define NET_EVENT_L4_IPV4_CONNECTED			\
```

Comment:
Could you also add these new events to `subsys/net/lib/shell/events.c` so that the event monitor can show user friendly information about them.